staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -641,11 +642,22 @@ func (qs *queueSet) dispatchLocked() bool {
 	return ok
 }
 
-// cancelWait ensures the request is not waiting.  This is only
+// cancelWaitWithAdditionalLatency ensures the request is not waiting.  This is only
 // applicable to a request that has been assigned to a queue.
-func (qs *queueSet) cancelWait(req *request) {
+func (qs *queueSet) cancelWaitWithAdditionalLatency(req *request) {
 	qs.lock.Lock()
 	defer qs.lock.Unlock()
+	if req.executed {
```

Comment:
Do we need the "executed" field?

I started wondering (didn't check that deeply enough yet) if we can't check that based on "decision" field being set to decisionExecute...